tests/integration/test_apt.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from urllib.request import urlopen
+
+from charms.operator_libs_linux.v0 import apt
+from helpers import get_command_path
+
+logger = logging.getLogger(__name__)
+
+
+def test_install_package():
+    try:
+        apt.update()
+        apt.add_package("zsh")
+        apt.add_package(["golang-cfssl", "jq"])
+    except apt.PackageNotFoundError:
+        logger.error("A specified package not found in package cache or on system")
+    except apt.PackageError as e:
+        logger.error(f"Could not install package. Reason: {e.message}")
+
+    paths = [get_command_path(c) for c in ["zsh", "cfssl", "jq"]]
+    assert paths == ["/usr/bin/zsh", "/usr/bin/cfssl", "/usr/bin/jq"]
```

Comment:
I guess that strictly speaking, we're not 100% certain that our libraries installed these apps. If any of them make it into the distro by default in the future, our tests will cease to test.

I like the package selection here -- useful, well maintained libraries that are nonetheless probably not going to make it into the default ubuntu image in the forseeable. I think that I'm good with this.